src/geometric_transformations.rs

```diff
@@ -710,11 +710,11 @@ where
 
 #[inline(always)]
 fn interpolate_nearest<P: Pixel>(image: &Image<P>, x: f32, y: f32, default: P) -> P {
-    let rx = x.round();
-    let ry = y.round();
+    let rx = (x + if x.is_sign_positive() { 0.5 } else { -0.5 }) as i32;
+    let ry = (y + if y.is_sign_positive() { 0.5 } else { -0.5 }) as i32;
 
     let (width, height) = image.dimensions();
-    if rx < 0f32 || rx >= width as f32 || ry < 0f32 || ry >= height as f32 {
+    if rx < 0 || rx >= width as i32 || ry < 0 || ry >= height as i32 {
```

Comment:
What if width or height is larger than i32::MAX?

Reply:
Oh thanks for pointing that out I didn't consider that before. I just fixed it by casting width/height to u32 instead of casting rx/ry to i32 and reordering it to make sure they aren't less than 0 first.